packages/@aws-cdk/aws-s3/lib/bucket.ts

```diff
@@ -1692,6 +1719,23 @@ export class Bucket extends BucketBase {
       };
     });
   }
+
+  private getOrCreateAutoDeleteObjectsResource() {
+    return CustomResourceProvider.getOrCreate(this, AUTO_DELETE_OBJECTS_RESOURCE_TYPE, {
+      codeDirectory: path.join(__dirname, 'auto-delete-objects-handler'),
+      runtime: CustomResourceProviderRuntime.NODEJS_12,
+      policyStatements: [
+        {
+          Effect: 'Allow',
+          Resource: '*',
```

Comment:
```suggestion
          Resource: this.bucketArn,
```

Reply:
It turns out this policy for this lambda needs to have access to all buckets in the stack, since it is a singleton that is shared with the entire stack. So I think specifying either * or arn:aws:s3:::* should work here.

Reply:
I actually think this is a very important consideration we should think about. I'm not comfortable with creating a lambda function that has permissions to delete all objects from all buckets. 

I think that it's better to create a provider per bucket, which i know is currently not possible, but maybe we should add it. 

@rix0rrr What do you think about either making the constructor public, or allow passing a scope?

Reply:
Can't we use a bucket policy here instead?